go-controller/extensions/pkg/node/node_windows.go

```diff
@@ -24,14 +57,76 @@ func NewNodeHandler(clientset kubernetes.Interface, nodeName string) (types.Node
 // Add function learns about a new node being added to the cluster
 // For a windows node, this means watching for all nodes and programming the routing
 func (n *nodeController) Add(node *kapi.Node) {
+	network, err := hcn.GetNetworkByID(n.networkID)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	nodeSubnet, ok := node.Annotations[extensionHostSubnet]
+
+	if !ok {
+		log.Errorf("Can't retreive the host subnet from the '%s' node's annotations", node.Name)
+		return
+	}
+
+	nodeIPAddress := getFirstInternalIp(node.Status.Addresses)
+
+	if nodeIPAddress == nil {
+		log.Errorf("Can't retreive node '%s' internal IP address from the node's annotations", node.Name)
+		return
+	}
+
+	drmac, ok := node.Annotations[extensionDrMac]
+
+	if !ok {
+		log.Errorf("Can't retreive the distributed router MAC address from the '%s' node's annotations", node.Name)
+		return
+	}
+
+	networkPolicySettings := hcn.RemoteSubnetRoutePolicySetting{
+		IsolationId:                 virtualNetworkIdentifier, // VXLAN virtual network Identifier. Is expected to be 4097 or higher on Windows
+		DistributedRouterMacAddress: drmac,                    // Distributed router/gateway MAC address
+		ProviderAddress:             *nodeIPAddress,           // Host IP address of the node
+		DestinationPrefix:           nodeSubnet,
+	}
+
+	err = AddRemoteSubnetPolicy(network, &networkPolicySettings)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
 	return
 }
 
 func (n *nodeController) Update(oldNode, newNode *kapi.Node) {
+	// TODO: Is this the right approach? In which case is Update() called?
```

Comment:
@JocelynBerrendonner Update will be called for a node whenever anything in the node object changes, which will usually be completely unrelated changes. So doing a delete+add will cause a lot of churn. Check if the annotations are still valid (eg they exist and are well formatted) and if not, call Delete(). If they are well-formatted, then ideally we figure out of they have changed (based on cached info somewhere or reading existing RemoteSubnetPolicy?) and only update in that case.

Linux side needs a bit of work here too, but it doesn't do quite as much as the Windows side it looks like so the hit is OK.

Reply:
I added a logic to compare the "oldnode" and "newnode" and only delete/re-add the node if the subnet, IP, or DRMAC on the new node has changed. Let me know if that is what you had in mind!